cli/logout.go

```diff
@@ -41,25 +42,46 @@ func (l *LogoutCommand) runLogout(args []string) error {
 	}
 
 	// TODO: better way to get registry server address, or get it from a default variable
-	registry := serverAddress
-	if registry == "" {
-		registry = "https://index.docker.io"
+	var (
+		registry = serverAddress
+		err      error
+	)
+
+	if registry, err = getRegistry(l.cli, registry); err != nil {
+		// error appears only in get default registry
+		fmt.Fprintf(os.Stderr, "Fail to get default registry: %s\n", err)
 	}
 
-	if !credential.Exist(serverAddress) {
+	if !credential.Exist(registry) {
 		fmt.Fprintf(os.Stdout, "Has not logged in registry: %s\n", registry)
 		return nil
 	}
 
-	if err := credential.Delete(serverAddress); err != nil {
+	if err := credential.Delete(registry); err != nil {
 		fmt.Fprintf(os.Stderr, "Fail to remove login credential: %s\n", err)
 		return err
 	}
 
-	fmt.Fprintf(os.Stdout, "Remove login credential for registry:%s\n", registry)
+	fmt.Fprintf(os.Stdout, "Remove login credential for registry: %s\n", registry)
 	return nil
 }
 
+func getRegistry(cli *Cli, registry string) (string, error) {
```

Comment:
I am not sure I like all the design of this function, to be honest. Maybe we could just use the implementation of this function directly in the place where to call this function. It is some kind of strange to input a cli parameter in this function. This will break the `object-oriented`. Maybe I am not exactly correct. But correct me if was not on the right way.

Reply:
how about remove cli as parameter.

Reply:
maybe we could use a function `func (cli *Cli) getRegistry(registry string) (string, error)`

or just use the implementation of this function directly in the place where to call this function